pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -461,7 +461,7 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 
 			case RouteType:
 				routes := discReq.GetResourceNames()
-				if len(routes) == len(con.Routes) || len(routes) == 0 {
+				if reflect.DeepEqual(routes, con.Routes) || len(routes) == 0 {
```

Comment:
maybe we need to sort these two arrays

Reply:
and maybe do the check on `len(routes)` first if you are about to do deep equal as it may be true and save the reflection time.

Reply:
right, have done :-) PTAL